test/viewmodels/more_viewmodel_test.dart

```diff
@@ -0,0 +1,185 @@
+        startDate: null,
+        startDateCancellationWithoutRefundNewStudent: null,
+        startDateCancellationWithRefund: null,
+        startDateRegistration: null,
+        endDate: null,
+        endDateCourses: null),
+  ];
+
+  final List<CourseActivity> coursesActivities = [
+    CourseActivity(
+        courseName: 'sample course',
+        courseGroup: '',
+        activityName: '',
+        activityDescription: '',
+        activityLocation: '',
+        startDateTime: null,
+        endDateTime: null),
+  ];
+  group('MoreViewModel - ', () {
+    setUp(() async {
+      cacheManagerMock = setupCacheManagerMock() as CacheManagerMock;
+      settingsManagerMock = setupSettingsManagerMock() as SettingsManagerMock;
+      courseRepositoryMock =
+          setupCourseRepositoryMock() as CourseRepositoryMock;
+      preferenceService = setupPreferencesServiceMock();
+      userRepositoryMock = setupUserRepositoryMock() as UserRepositoryMock;
+      navigationService = setupNavigationServiceMock();
+      githubApiMock = setupGithubApiMock() as GithubApiMock;
+      setupLogger();
+      appIntl = await setupAppIntl();
+
+      viewModel = MoreViewModel(intl: appIntl);
+
+      CourseRepositoryMock.stubSessions(courseRepositoryMock,
+          toReturn: sessions);
+      CourseRepositoryMock.stubCoursesActivities(courseRepositoryMock,
+          toReturn: coursesActivities);
+    });
+
+    tearDown(() {
+      unregister<CacheManager>();
+      unregister<SettingsManager>();
+      unregister<CourseRepository>();
+      unregister<PreferencesService>();
+      unregister<UserRepository>();
+      unregister<NavigationService>();
+      unregister<GithubApi>();
+    });
+
+    group('logout - ', () {
+      test('If cache manager is emptied', () async {
+        await viewModel.logout();
+        verify(cacheManagerMock.empty());
+      });
+
+      test('If preference manager is clear', () async {
+        await viewModel.logout();
+        verify(preferenceService.clear());
+      });
+
+      test('If user repository is logOut is called', () async {
+        await viewModel.logout();
+        verify(userRepositoryMock.logOut());
+      });
+
+      test(
+          'If settings manager has reset language and theme mode and has notified listener',
+          () async {
+        await viewModel.logout();
+        verify(settingsManagerMock.resetLanguageAndThemeMode());
+        verify(settingsManagerMock.notifyListeners());
+      });
+
+      test('If the cache for courses activities and sessions is cleared',
+          () async {
+        await viewModel.logout();
+        expect(courseRepositoryMock.sessions.length, 0);
+        expect(courseRepositoryMock.coursesActivities.length, 0);
+      });
+
+      test('If the navigationService is rerouted to login', () async {
+        await viewModel.logout();
+        verify(navigationService.pop());
+        verify(navigationService.pushNamedAndRemoveUntil(RouterPaths.login));
+      });
```

Comment:
All of these could be one and only test. When you use `expect(...)` you can add a string as `reason:` (why this should be expected).

Also, we are missing some cases, what happens if the cache manager, settings manager, ... throw an error? 😄